### ISSUE TITLE
flutter doctor immediately closes without any output

### ISSUE THREAD

**Issue by mobynet1** [flutter/flutter#156922]
### Steps to reproduce

I am trying to get Flutter and Dart configured on a Windows 11 development box. I am following the flutter installation documentation and when I run the 'flutter doctor', which is one of the early installation verification steps, the cmd windows simply gets closed without any output. Kind of a lame utility if it does nothing but abort.

Following the documentation, I downloaded the Windows Flutter zip file and placed it into a simple folder, D:\Development\Flutter. I already have git installed (git version 2.46.1.windows.1 which I get from running the `git --version` command). Git is in my path as is the flutter folders. Either this utility is broken (I can't even get this flutter thing to install!!!)

I am working on a project that it seems as though every avenue has a blocker. dotnet gRPC failed because it could not handle UDP sockets so I had to throw out tons of C#/.NET gRPC code and replace it with go and libp2p. I was trying to use .NET Maui, but the Maui team seems unable to produce anything of quality. I reported bugs over two years ago that are still open. I replaced Maui with Uno, and since it is tied heavily to WinUI 3, it will suffer from the low quality of the WinUI/.NET code. Even Windows is getting dumber by the day where simple things do not work any longer.

Visual Studio has bugs that prevent anything written in xaml with a code-behind file from compiling unless you compile it for the base windows configuration - which is completely unacceptable when writing cross-platform code.

In a nutshell, Microsoft and the rapidly deteriorating quality of everything they touch, has made developing cross-platform apps using .NET tools or Visual Studio 2022 virtually impossible. 

I am on a brand-new Windows installation and am simply installing the software I need to do my job. I have been in blocker hell for nine months trying every workaround known to man and cannot get their crap to work...so I decided to give Flutter a whirl.

So I must say that I am SEVERELY disappointed that I cannot even get a base installation of Flutter to work using YOUR 'Get Started' information. What gives?!?! It is as if the software gods are all against me!! Maybe what it is saying is that I need to get my resume updated so I can go apply for that Home Depot or Walmart greeter job because this software gig is going to kill me.

Can someone please give me an explanation as to why the base flutter installation documentation does not provide the correct instructions for getting a Windows dev environment configured? I have been a software engineer for over 40 years so I don't think I am stupid. What gives y'all??? Please help me get flutter installed... per YOUR instructions!!

### Expected results

I just want the damned thing to work!!

### Actual results

nada...

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

![image](https://github.com/user-attachments/assets/8a39cc3d-f50b-4c91-9700-d5ef004a82be)

This was from a powershell window. When I run this in a plain cmd window, the Flutter Doctor command simply aborts with no output...

**Comment by christopherfujino** [comment]
It looks like the logic for this is actually trying to invoke `git rev-parse HEAD`, rather than checking if git is in your PATH:

https://github.com/flutter/flutter/blob/main/bin/internal/shared.bat#L55-L64

What happens if you run `git rev-parse HEAD` in your Flutter repo?

**Comment by darshankawar** [comment]
@mobynet1 

Based on the error you are getting `Error: Unable to find git in your PATH`, it may be similar to https://github.com/flutter/flutter/issues/123161 that you can check.




> What happens if you run `git rev-parse HEAD` in your Flutter repo?

Please provide these info as well.

**Comment by mobynet1** [comment]
@darshankawar At present, I have no Flutter projects or repos, so I don't think that #123161 applies here. I also own the Flutter installation folder and have full control over it and its subfolders. Since I was only doing a `flutter doctor` with no flutter repo even created, where is there any reference to a repo that could possibly cause the `unsafe` error to manifest?

Just to be sure, I tried to run `flutter doctor` inside the folder that I _intend_ to create a repo and it still aborts.

In short, it could have something to do with what git is trying to do, but what that `it` is, I don't have a clue. I am suspect of that being the cause. There has to be something else at play here.

**Comment by christopherfujino** [comment]
> It looks like the logic for this is actually trying to invoke `git rev-parse HEAD`, rather than checking if git is in your PATH:
> 
> https://github.com/flutter/flutter/blob/main/bin/internal/shared.bat#L55-L64
> 
> What happens if you run `git rev-parse HEAD` in your Flutter repo?

By "your flutter repo", I mean the Flutter SDK installation (this is the directory the batch script is attempting and failing to run `git rev-parse HEAD` in)

**Comment by mobynet1** [comment]
@christopherfujino I get this: `2663184aa79047d0a33a14a3b607954f8fdd8730`

**Comment by christopherfujino** [comment]
> @christopherfujino I get this: `2663184aa79047d0a33a14a3b607954f8fdd8730`

Hmm, which is a valid Flutter SDK stable release commit: https://github.com/flutter/flutter/commit/2663184aa79047d0a33a14a3b607954f8fdd8730

this looks like a bug in our windows batch file.

To further help us diagnose the bug, can you share the output of:

```
Where flutter
```

And verify this path is the same one you just ran `git rev-parse HEAD` on?

Oh, and also:

```
Where git
```

And verify this is the same terminal environment that you are *not* able to run `flutter doctor` from (if for example, you run the terminal in VSCode or Visual Studio, those may add additional paths to the PATH). Thanks!

**Comment by bkonyi** [comment]
Hi @mobynet1, have you managed to get this resolved?

**Comment by mobynet1** [comment]
Guys, I don't know what you are looking for that I did not provide. The above says everything you need to know.

1) I was doing a fresh install on a machine that never had flutter or dart installed.
2) Following your 'Get Started' documentation, at some point after making sure that the prereq's were installed, I simply ran flutter doctor and it just vanished... Look at the screenshot provided!
3) Git IS installed! That is visible in the screenshot.
4) flutter doctor says git is NOT in the path...but it plainly is... THAT IS THE ERROR!!

This is nothing that I did or should have to do. If I run your utilities, they should work or at a minimum provide better diagnostic error messages. Your flutter doctor did not see git in the path when it clearly is. THAT IS WHAT NEEDS ATTENTION!

**Comment by bkonyi** [comment]
Hi @mobynet1,

This issue was automatically closed after there was no response from you for nearly a month after we requested more information.

We understand that you have `git` installed and it's on your path, but that's not enough information for us to make progress on debugging your issue. We have hundreds of thousands of other users, many of which are using Windows, who aren't encountering the same issue as you so there's likely some edge case that we're encountering on your particular environment that we can't reproduce or investigate without your help.

As @christopherfujino pointed out, this is likely not necessarily a `flutter doctor` issue but an issue with `flutter.bat`, which is a wrapper script that downloads artifacts needed for your particular OS and compiles the actual Flutter CLI tool. This error message is coming from [here](https://github.com/flutter/flutter/blob/master/bin/internal/shared.bat#L66) after we try and run `git rev-parse HEAD` and fail to get a result. If you could add `echo %PATH%` to the line before and share the output, we can figure out whether or not the script is executing in the environment that we expect it to be.

Also, as @christopherfujino requested in his last comment, please provide the output of `where flutter` and `where git` from the terminal you're encountering this issue in which will hopefully give us some more clues as to what's going on here.

Finally, I'd like to remind you to take a moment and read [Flutter's Code of Conduct](https://github.com/flutter/flutter/blob/master/CODE_OF_CONDUCT.md). We're here to help, but inflammatory or aggressive responses, no matter how frustrating an issue may be, don't help us help you.

**Comment by bkonyi** [comment]
> 2. I provided enough information for you to test this on your own. The Microsoft MAUI team always wants the end users to do their work debugging their code. This is the same here in this situation. Your product is broken and I opened an issue.

You did not, which is why we asked for more information to help us debug the issue. If this was a widespread issue, we would be flooded with issues reporting this problem. That tells me that there's something specific about your setup that is causing the issue, especially since the script is failing so early in its execution on a command that isn't part of our actual tech stack.

> 3. I expect vendors to do their own debugging. I told you that I had a clean VM with nothing installed. I followed your instructions to the letter and flutter doctor failed. Why can't you simply follow my report to see what is going on? Again, I do not have time to debug your product!

We didn't ask you to fix the issue yourself, but we are not oracles with intimate knowledge of your system's configuration. We regularly ask issue reporters for additional information so we can get enough context to figure out what's going on.

> 4. `where flutter` and `where git` both found the paths. The `flutter doctor` could not find git. I closed command windows and reopened them to ensure any and all environment variables were set properly.

It's much more helpful to us if you share the output of these commands.

> 5. Here's my PATH variable:
>    `/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/steve/.dotnet/tools:/usr/local/go/bin:/usr/lib/dotnet:/home/steve/dotnet:/home/steve/development/flutter/bin:'

Thank you for sharing your PATH, it provides some more clues as to what's going on. This is a Unix styled PATH, not one I would expect to see on Windows.

Do you have WSL installed or `git bash`? If so, I'm wondering if your terminal is running `flutter` (the Bash script) instead of `flutter.bat`. This could explain why `where git` gives an actual result when run from your Powershell instance but it isn't found by the `flutter` script. Can you try running `flutter.bat doctor` or `D:\Development\Flutter\bin\flutter.bat doctor` and see if that works? 

> Trying to get Dart/Flutter installed should be seamless and straight forward, but it wasn't. If I download something from GitHub or a vendor, it should not be a stretch to expect it to work out of the box.

I agree 100%, this should just work out of the box and that's the experience we strive to provide. I understand that this is frustrating but, unfortunately, software engineering is hard and it's impossible to predict and handle every edge case, particularly when you're shipping code to run on hundreds of thousands of slightly different but unique environments.

> I reported an issue that I felt you would be interested in, but if not that does not bode well about your attitude towards your product's quality. I will stop reporting issues if you are not invested enough to try to do the research necessary to get an issue resolve.

I don't see where we've indicated in this thread that we're not interested in helping you resolve this issue given we've responded to each of your comments. I'm happy to try and help you out here, but it will require some back and forth discussion to resolve. If not, we can leave this issue closed.

**Comment by mobynet1** [comment]
I am doing my development on a Linux (Ubuntu 24.04) VM...

**Comment by atang08** [comment]
Hello, I am on the same boat and its been a good 8 hours I spent trying to get this up and running.

Starting to think this could be a bug.  Flutter location extracted to: C:\flutter

I'm using Windows 11 64 bit, and followed the installation steps and still get this message when running flutter doctor.  

`Error: Unable to find git in your PATH.`

Like the OP, when I used the regular command prompt, it just spits out the error and closes the window.   If I use PowerShell 5.1, it just shows the error and a new prompt, window stays open.

I have git installed and running where git shows:
`C:\flutter\bin\mingit\mingw64\bin\git.exe`
(I tried every git path possible...I tried this one along with:
```
C:\Program Files\Git\bin
C:\Program Files\Git\cmd
C:\flutter\bin\mingit\cmd
```

I downloaded the most up to date git windows 64 bit version (2.42) and everything seems to be fine.

When running the git rev-parse HEAD:
```
PS C:\flutter> git rev-parse HEAD
dec2ee5c1f98f8e84a7d5380c05eb8a3d0a81668
```
Running where flutter:
[C:\flutter\bin\flutter
C:\flutter\bin\flutter.bat]

For my User's windows path environment variable, it includes:
flutter bin, git paths.

I don't believe this is a Path issue now and something to do with flutter.

**Comment by bkonyi** [comment]
> Hello, I am on the same boat and its been a good 8 hours I spent trying to get this up and running.
> 
> Starting to think this could be a bug. Flutter location extracted to: C:\flutter
> 
> I'm using Windows 11 64 bit, and followed the installation steps and still get this message when running flutter doctor.
> 
> `Error: Unable to find git in your PATH.`
> 
> Like the OP, when I used the regular command prompt, it just spits out the error and closes the window. If I use PowerShell 5.1, it just shows the error and a new prompt, window stays open.
> 
> I have git installed and running where git shows: `C:\flutter\bin\mingit\mingw64\bin\git.exe` (I tried every git path possible...I tried this one along with:
> 
> ```
> C:\Program Files\Git\bin
> C:\Program Files\Git\cmd
> C:\flutter\bin\mingit\cmd
> ```
> 
> I downloaded the most up to date git windows 64 bit version (2.42) and everything seems to be fine.
> 
> When running the git rev-parse HEAD:
> 
> ```
> PS C:\flutter> git rev-parse HEAD
> dec2ee5c1f98f8e84a7d5380c05eb8a3d0a81668
> ```
> 
> Running where flutter: [C:\flutter\bin\flutter C:\flutter\bin\flutter.bat]
> 
> For my User's windows path environment variable, it includes: flutter bin, git paths.
> 
> I don't believe this is a Path issue now and something to do with flutter.
> 
> I also ran `flutter.bat doctor` with the same results.

Thanks for the update and details @huycozy! That's very helpful!

Since this issue was closed by the reporter, let's continue this investigation over on #159018.

**Comment by huycozy** [comment]
@bkonyi I think you meant @atang08 😄. I only quoted and linked the new issue to this issue.

**Comment by bkonyi** [comment]
> @bkonyi I think you meant @atang08 😄. I only quoted and linked the new issue to this issue.

Yes I did 😅 oops!

**Comment by bkonyi** [comment]
This should be resolved by https://github.com/flutter/flutter/pull/159424.
